artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireProtocolManager.java

```diff
@@ -236,6 +238,18 @@ public void removeConnection(ConnectionInfo info, Throwable error) throws Invali
       }
    }
 
+   /*** if set, the OpenWire connection will bypass the tcpReadBuferSize and use this value instead.
+    *   This is by default -1, and it should not be used unless in extreme situations like on a slow storage. */
+   public int getMaxActorSize() {
```

Comment:
this does not make sense to me, a config that should not be used. The intent is to bound the queue and it is matched to the receive buffer size. if it needs to be lower or higher then modify the acceptor receive buffer size. The bounding of the actor stops reading, so at a max the receive buffer size will be pending on the socket and in the broker. Then the client will block on a tcp write. When will it make sense to have more than send buffer pending on the broker? I don't know how to reason about setting a maxActorSize on the protocol manager!
in terms of a name, it could be actorPendingQueueMaxSizeBytes but that is an aside. Only add that if we know when it should be set.

Reply:
@gtully if you have a slow device to sync, and you read say 1M worth of data, but take more than the configured TTL to read again your clients will start to fall apart.

This is a backup plan in case this scenario happens. By default it should use the tcp-read-buffer, but when things fall apart you have the option to use a lower read-value. I have seen that happening on my tests and I don't want to take the risk without having a backup plan.